utils/virtual_machine/aws_provider.py

```diff
@@ -75,7 +80,7 @@ def pulumi_start_program():
     def _start_vm(self, vm):
         # Check for cached ami, before starting a new one
         should_skip_ami_cache = os.getenv("SKIP_AMI_CACHE", "False").lower() == "true"
-        ami_id = self._get_cached_ami(vm) if not should_skip_ami_cache else None
+        ami_id = pulumi.Output.from_input(asyncio.to_thread(self._get_cached_ami, vm)) if not should_skip_ami_cache else None
```

Comment:
Can someone with more pulumi knowledge confirm that it is fine to use an `Output` here rather than the actual ami_id?

It seems to work, but would like some confirmation that it's using the expected ami.